# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1430463408 -7200
#      Fri May 01 08:56:48 2015 +0200
# Branch stable
# Node ID f0a8c1bc3b6083e58b881d92699b91b1efe771ac
# Parent  f8d443fe0de3b5e4d0c3468d0fcdfe605fa57936
fix checking copy shortcut in terminal when no setting available (bug #44968)

* QTerminal.cc (notice_settings): check for empty copy shortcut from settings
  file and take the default in this case

diff --git a/libgui/qterminal/libqterminal/QTerminal.cc b/libgui/qterminal/libqterminal/QTerminal.cc
--- a/libgui/qterminal/libqterminal/QTerminal.cc
+++ b/libgui/qterminal/libqterminal/QTerminal.cc
@@ -134,16 +134,20 @@ QTerminal::notice_settings (const QSetti
   // check whether Copy shoretcut is Ctrl-C
   int set = settings->value ("shortcuts/set",0).toInt ();
   QKeySequence copy;
   QString key = QString ("shortcuts/main_edit:copy");
   if (set)
     key.append ("_1");  // if second set is active
   copy = QKeySequence (settings->value (key).toString ()); // the copy shortcut
 
+  // if copy is empty, shortcuts are not yet in the settings (take the default)
+  if (copy.isEmpty ())         // QKeySequence::Copy as second argument in
+    copy = QKeySequence::Copy; // settings->value () does not work!
+
   //  dis- or enable extra interrupt action
   QKeySequence ctrl;
   ctrl = Qt::ControlModifier;
 
   bool extra_ir_action = (copy != QKeySequence (ctrl + Qt::Key_C));
 
   _interrupt_action->setEnabled (extra_ir_action);
   has_extra_interrupt (extra_ir_action);
